inference-engine/src/mkldnn_plugin/mkldnn_node.cpp

```diff
@@ -1454,9 +1460,68 @@ std::vector<VectorDims> MKLDNNNode::shapeInferGeneric(const std::vector<Shape>&
 
         newOutputShapes[i] = partShape.get_shape();
     }
+
     return newOutputShapes;
 }
 
+std::vector<VectorDims> MKLDNNNode::shapeInferGeneric(const std::vector<Shape>& shapes,
+                                                      uint32_t input_value_port_mask) const {
+    // collect input shapes
+    std::vector<ov::StaticShape> input_shapes;
+    if (shapes.empty()) {
```

Comment:
You are right, but I have concern from performance perspective.

The path of  collect `shapes` vector and call `shapeInferGeneric(const std::vector<Shape>& shapes, uint32_t input_value_port_mask)` is not optimal, because we have to transform shapes from `VectorDims` to `Shape` and then to `StaticShape`, this overhead (of arguments forming) cannot be neglected since many shape inference functions are very light-weighted function by themselves.

Reply:
This problem can be addressed moving the actual implementation to the third utility function with `(const std::vector<ov::StaticShape>& shapes, uint32_t input_value_port_mask)` function in the `private` section, so both the protected interface calls can use that "underlying" implementation.

Reply:
OK, I see, changed.